packages/frontend/src/components/countdown.tsx

```diff
@@ -0,0 +1,106 @@
+'use client'
+import { UnixTime } from '@l2beat/shared-pure'
+import { memo, useState } from 'react'
+import { useEffect } from 'react'
+import { cn } from '~/utils/cn'
+
+interface Props {
+  expiresAt: number
+  size?: 'md' | 'sm'
+  className?: string
+}
+
+export function Countdown({ expiresAt, size = 'md', className }: Props) {
+  const [secondsLeft, setSecondsLeft] = useState(
+    expiresAt - UnixTime.now().toNumber(),
+  )
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setSecondsLeft((timeLeft) => timeLeft - 1)
+    }, 1000)
+    return () => clearInterval(interval)
+  }, [expiresAt])
+
+  const { months, days, hours, minutes, seconds } = getTimeParts(secondsLeft)
+
+  return (
+    <div
+      className={cn(
+        'flex w-max gap-x-1 rounded-lg border border-divider bg-surface-secondary p-2',
+        className,
+      )}
+    >
+      {months > 0 && (
+        <MemoizedTimePart suffix="mo" size={size}>
+          {months}
+        </MemoizedTimePart>
+      )}
+      <MemoizedTimePart suffix="d" size={size}>
+        {days}
+      </MemoizedTimePart>
+      <MemoizedTimePart suffix="hr" size={size}>
+        {hours}
+      </MemoizedTimePart>
+      <MemoizedTimePart suffix="m" size={size}>
+        {minutes}
+      </MemoizedTimePart>
+      {months === 0 && (
+        <MemoizedTimePart
+          suffix="s"
+          size={size}
+          className={cn(
+            size === 'md' && 'w-[67px]',
+            size === 'sm' && 'w-[54px]',
+          )}
+        >
+          {seconds}
+        </MemoizedTimePart>
+      )}
+    </div>
+  )
+}
+
+const MemoizedTimePart = memo(TimePart)
+function TimePart({
+  children,
+  suffix,
+  className,
+  size = 'md',
+}: {
+  children: React.ReactNode
+  suffix: string
+  className?: string
+  size?: 'md' | 'sm'
+}) {
+  return (
+    <div
+      className={cn(
+        'rounded bg-brand px-3 py-2 text-center font-bold text-primary-invert',
+        size === 'md' && 'text-[28px]',
+        size === 'sm' && 'text-[18px]',
+        className,
+      )}
+    >
+      <span>{children}</span>
+      <span
+        className={cn(
+          'ml-0.5',
+          size === 'md' && 'text-lg leading-none',
+          size === 'sm' && 'text-xs leading-none',
+        )}
+      >
+        {suffix}
+      </span>
+    </div>
+  )
+}
+
+function getTimeParts(timeLeft: number) {
+  const months = Math.floor(timeLeft / (60 * 60 * 24 * 30))
```

Comment:
LOL

Reply:
That's definitely wrong. Today is Dec 13th. If I see a countdown of 1 mo 2 days I expect Jan 15th. But december has 31 days so this countdown will actually expire at Jan 14th

Reply:
Yeah, I know, forgot to do smth with it